api-tests/dev_apis/crypto/test_c021/test_data.h

```diff
@@ -110,7 +110,7 @@ static const test_data check1[] = {
  PSA_KEY_DERIVATION_INPUT_SECRET, 224, {0}, 0,
  PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1), 224, PSA_KEY_USAGE_EXPORT,
  PSA_ALG_ECDSA_ANY,
- PSA_ERROR_INVALID_ARGUMENT
```

Comment:
taken this patch for testing, the return value is still INVALID_ARGUMENT instead of NOT_SUPPORTED, I might have to check with tfm team about the mbedtls version they are using(tfm 1.4 version) currently.

Reply:
TFM uses this version of mbedtls(found from build log) : HEAD is now at 8df2f8e7 Merge pull request #842 from ARMmbed/mbedtls-3.0.0rc0-pr

Reply:
in psa_crypto.c, function : psa_generate_derived_key_internal
the memory allocation is failing as shown below

![image](https://user-images.githubusercontent.com/46363970/147106657-a41fb8cb-9b5e-478f-aadf-ac695c8665c2.png)

the watch variable data
![image](https://user-images.githubusercontent.com/46363970/147106779-fd3c042f-3f3b-43e7-9fd0-0caa9f427dbe.png)

Reply:
@gilles-peskine-arm please suggest whats next?

Reply:
@gilles-peskine-arm 
please find the below findings

**test key type input** : PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1)  --> 0x7112

in **psa_crypto.c** file, function **psa_generate_derived_key_internal**, there is check for key type raw
if( ! key_type_is_raw_bytes( slot->attr.type ) )  --> key_type_raw_bytes returned 0, so the following code executed.
![image](https://user-images.githubusercontent.com/46363970/147120289-e688ebd9-d602-46e7-835c-27ee63a67122.png)

could you please suggest something here.

Reply:
@gilles-peskine-arm , could you please take a look at the above comment, if these things are handled in latest mbedtls 3.1.0 then i can merge it.

Reply:
Sorry, this had slipped off my radar. Mbed TLS still returns `PSA_ERROR_INVALID_ARGUMENT` in this case. We're going to fix this as part of https://github.com/ARMmbed/mbedtls/pull/5139.

Reply:
@jk-arm Mbed-TLS/mbedtls#5139 was merged. #312 contains the fixes needed in psa-arch-tests. Would you like us to  to force-push this PR, or would you be fine with closing this one, and merging #312?